app/models/manageiq/providers/autosde/inventory/parser/storage_manager.rb

```diff
@@ -107,4 +109,10 @@ def cloud_volumes
       )
     end
   end
+
+  def storage_host_wwpn_candidates
+      collector.storage_hosts_wwpn_candidates.each do |candidate|
+        persister.storage_hosts_wwpn_candidates.build(**candidate)
```

Comment:
I recommend having your collector just return the `candidate` like:

```ruby
def storage_hosts_wwpn_candidates
  @storage_hosts_wwpn_candidates ||= @manager.autosde_client.StorageHostWWPNCandidatesApi
                                             .storage_hosts_wwpn_candidates_get
end
```

then your parser would look like
```ruby
def storage_host_wwpn_candidates
  collector.storage_hosts_wwpn_candidates.each do |candidate|
    persister.storage_hosts_wwpn_candidates.build(
      :candidates       => candidate.wwpn,
      :ems_ref          => candidate.system_uuid,
      :physical_storage => persister.physical_storages.lazy_find(candidate.system_uuid)

    )
  end
end
```